framework/validators/Validator.php

```diff
@@ -415,4 +419,30 @@ public function isEmpty($value)
             return $value === null || $value === [] || $value === '';
         }
     }
+
+    /**
+     * Returns cleaned attribute names with "!" characters in the beginning removed
+     * @return array
+     * @since 2.0.12
+     */
+    public function getAttributeNames()
+    {
+        return $this->attributeNames;
+    }
+
+    /**
+     * Saves attribute names with "!" characters in the beginning removed as an array
```

Comment:
How about

<code>Saves attribute names without \`!\` character at the beginning</code>